addons/mail/models/mail_thread.py

```diff
@@ -917,10 +921,11 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         ]
         # Delivered-To is a safe bet in most modern MTAs, but we have to fallback on To + Cc values
         # for all the odd MTAs out there, as there is no standard header for the envelope's `rcpt_to` value.
-        rcpt_tos_localparts = [
-            e.split('@')[0].lower()
-            for e in tools.email_split(message_dict['recipients'])
-        ]
+        rcpt_tos_localparts = []
+        for e in tools.email_split(message_dict['recipients']):
```

Comment:
e -> email_to ?

Reply:
Struggled with one test to understand why breaking and not every time... `email_to` was already used above :man_facepalming: using `recipient` instead